x/tx/signing/aminojson/encoder.go

```diff
@@ -81,6 +81,17 @@ func nullSliceAsEmptyEncoder(enc *Encoder, v protoreflect.Value, w io.Writer) er
 	}
 }
 
+// cosmosInlineJson replicates the behavior at:
+// https://github.com/CosmWasm/wasmd/blob/08567ff20e372e4f4204a91ca64a371538742bed/x/wasm/types/tx.go#L20-L22
+func cosmosInlineJson(_ *Encoder, v protoreflect.Value, w io.Writer) error {
+	blob, err := json.RawMessage(v.Bytes()).MarshalJSON()
```

Comment:
this is a lot of machinery to just print a pre-rendered string, `v.Bytes()` doesn't even need to be valid json.

```
	raw := json.RawMessage("foo")
	bz, err := raw.MarshalJSON()
	fmt.Println(string(bz))
    => "foo"
```

Given this, could this encoder just be called `bytes_as_string` and include a type check on `v protoreflect.Value` (it should be bytes) similar to `nullSliceAsEmptyEncoder` checks that `v` is a List?

Reply:
All of this PR is specific to Amino **JSON** and embedded somewhere in a JSON document. So I wonder why the emphesis on arbitrary bytes here is useful.

I don't think the naming choice "bytes_as_string" is great, because to me it sounds like adding a JSON string to the document instead of a nested JSON object. I.e. from the name it could easily mean that the bytes `{"foo":123}` becomes

```json
{
  "envelope":"{\"foo\":123}"
}
```
instead of

```json
{
  "envelope": {
    "foo": 123
  }
}
```

We need the later here, which is why I suggested inline JSON.

Reply:
I see what you mean, and if there is a possible ambiguity, let's indeed reword it.

Reply:
Great, will take a look.

Do you have an idea how I can create a simple test that shows how the field is embedded in a larger structure (similar to shown in my comment)?

Reply:
Yes, like this: https://github.com/cosmos/cosmos-sdk/blob/41f9272/x/tx/signing/aminojson/json_marshal_test.go#L213-L329

Reply:
Great, thank you. Will do that during or after Easter depending on family mood

Reply:
Thank again! Here we go: https://github.com/cosmos/cosmos-sdk/pull/19919